task16/index.js

```diff
@@ -0,0 +1,285 @@
+      "article": [
+        {
+          "data": "2022/02/28",
+          "title": "Sports01",
+          "comment": [
+            {
+              "name": "Ginny",
+              "data": "2022/02/24"
+            },
+            {
+              "name": "Morry",
+              "data": "2022/02/25"
+            },
+            {
+              "name": "Percy",
+              "data": "2022/02/24"
+            }
+          ]
+        },
+        {
+          "data": "2022/03/01",
+          "title": "Sports02",
+          "comment": [{}]
+        },
+        {
+          "data": "2022/03/03",
+          "title": "Sports03",
+          "comment": [{}]
+        },
+        {
+          "data": "2022/03/03",
+          "title": "Sports04",
+          "comment": [{}]
+        }
+      ]
+    }
+  ]
+
+const tabWrapper = document.getElementById("js-tabWrapper");
+const tab = document.getElementById("js-tabMenu");
+
+const createElementWithClassName = (element, name) => {
+  const createdElement = document.createElement(element);
+  createdElement.classList.add(name);
+  return createdElement;
+};
+
+const addLoading = () => {
+  const loadingPlace = createElementWithClassName("div", "loader");
+  loadingPlace.id = "loadingPlace";
+  tabWrapper.appendChild(loadingPlace);
+};
+
+const removeLoading = () => document.getElementById("loadingPlace").remove();
+
+const getFetchData = async () => {
+  try {
+// const respons = await fetch(URL);
+    // if (!respons.ok) {
+    //   throw new Error("ただいまサーバー側で通信が壊れています");
+    // }
+    // const json = await respons.json();
+    // return json;
+    //*myjsonが繋がらないので一時的に下記の記述をしています。
+    const json =  URL;
+    return json;
+  } catch (error) {
+    console.error("error:ただいまサーバー側で通信が壊れています");
+  }
+};
+
+const init = async () => {
+  addLoading();
+  try {
+    const value = await getFetchData();
+    renderTabMenu(value);
+    renderTabContainer(value);
+    // tabSwitch();
+  } catch (error) {
+    console.error(error);
+    tab.textContent = "データを読み込めませんでした";
+  } finally {
+    removeLoading();
+  }
+};
+init();
+
+const renderTabMenu = (value) => {
+  const tabMenuFragment = document.createDocumentFragment();
+  for (let i = 0; i < value.length; i++) {
+    const menuList = createElementWithClassName("li", "menu-list");
+    menuList.dataset.index = i;
+    menuList.textContent = value[i].category;
+    value[i].display === true && menuList.classList.add("active");
+    tabMenuFragment.appendChild(menuList);
+  }
+  tab.appendChild(tabMenuFragment);
+};
+
+const renderTabContainer = (value) => {
+  const tabContentsFragment = document.createDocumentFragment();
+  for (let i = 0; i < value.length; i++) {
+    const newsWrapper = createElementWithClassName("div", "news-wrapper");
+    const articleTitlelist = createElementWithClassName("ul", "article-title-list");
+    const newsImg = createElementWithClassName("img", "news-img");
+    value[i].display === true && newsWrapper.classList.add("show");
```

Comment:
IMO
```suggestion
    value[i].display && newsWrapper.classList.add("show");
```

Reply:
I thought I had to write true for it to show up. Thank you!!!
https://github.com/seeeen044/JavaScript_handson/commit/f7aaf72a5f2223e8e3fc6a3bb59a09d4d35bdb8b